bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/capability/SecurityCapability.java

```diff
@@ -0,0 +1,180 @@
+import org.openhab.binding.netatmo.internal.handler.ApiBridgeHandler;
+import org.openhab.binding.netatmo.internal.handler.CommonInterface;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SecurityCapability} is the base class for handler able to handle security features
+ *
+ * @author Gaël L'hopital - Initial contribution
+ *
+ */
+@NonNullByDefault
+class SecurityCapability extends RestCapability<SecurityApi> {
+    private final Logger logger = LoggerFactory.getLogger(SecurityCapability.class);
+
+    SecurityCapability(CommonInterface handler) {
+        super(handler);
+        ApiBridgeHandler bridgeApi = handler.getRootBridge();
+        if (bridgeApi != null) {
+            api = Optional.ofNullable(bridgeApi.getRestManager(SecurityApi.class));
+        }
+    }
+
+    @Override
+    protected void updateHomeData(HomeData homeData) {
+        NAObjectMap<HomeDataPerson> persons = homeData.getPersons();
+        NAObjectMap<HomeDataModule> cameras = homeData.getModules();
+        handler.getActiveChildren().forEach(handler -> {
+            HomeDataPerson dataPerson = persons.get(handler.getId());
+            if (dataPerson != null) {
+                handler.setNewData(dataPerson);
+            }
+            HomeDataModule data = cameras.get(handler.getId());
+            if (data != null) {
+                handler.setNewData(data);
+            }
+        });
+    }
+
+    @Override
+    protected void updateHomeStatus(HomeStatus homeStatus) {
+        NAObjectMap<HomeStatusPerson> persons = homeStatus.getPersons();
+        NAObjectMap<HomeStatusModule> cameras = homeStatus.getModules();
+        for (CommonInterface handler : handler.getActiveChildren()) {
+            HomeStatusPerson dataPerson = persons.get(handler.getId());
+            if (dataPerson != null) {
+                handler.setNewData(dataPerson);
+            }
+            HomeStatusModule dataCamera = cameras.get(handler.getId());
+            if (dataCamera != null) {
+                handler.setNewData(dataCamera);
+            }
+        }
+    }
+
+    @Override
+    protected void updateHomeEvent(HomeEvent homeEvent) {
+        String personId = homeEvent.getPersonId();
+        if (personId != null) {
+            handler.getActiveChildren().stream().filter(handler -> personId.equals(handler.getId())).findFirst()
+                    .ifPresent(handler -> handler.setNewData(homeEvent));
+        }
+        String cameraId = homeEvent.getCameraId();
+        handler.getActiveChildren().stream().filter(handler -> cameraId.equals(handler.getId())).findFirst()
+                .ifPresent(handler -> handler.setNewData(homeEvent));
+    }
+
+    Collection<HomeEvent> getCameraEvents(String cameraId) {
+        return api.map(api -> {
+            try {
+                return api.getCameraEvents(handler.getId(), cameraId);
+            } catch (NetatmoException | NoSuchElementException e) {
```

Comment:
NoSuchElementException can be thrown by what exactly ?

Reply:
Leftover. Corrected